openapi-codegen/core/src/main/scala/sttp/tapir/codegen/EndpointGenerator.scala

```diff
@@ -173,29 +196,62 @@ class EndpointGenerator {
       }
   }
 
-  private def ins(parameters: Seq[OpenapiParameter], requestBody: Option[OpenapiRequestBody])(implicit location: Location): String = {
+  private def ins(
+      parameters: Seq[OpenapiParameter],
+      requestBody: Option[OpenapiRequestBody],
+      endpointName: String,
+      targetScala3: Boolean,
+      jsonSerdeLib: JsonSerdeLib
+  )(implicit location: Location): (String, Option[String]) = {
+    def getEnumParamDefn(param: OpenapiParameter, e: OpenapiSchemaEnum, isArray: Boolean) = {
+      val enumName = endpointName.capitalize + strippedToCamelCase(param.name).capitalize
+      val queryParamRefs = if (param.in == "query") Set(enumName) else Set.empty[String]
+      val enumDefn = EnumGenerator.generateEnum(
+        enumName,
+        e,
+        targetScala3,
+        queryParamRefs,
+        jsonSerdeLib,
+        Set.empty,
+        isArray
+      )
+      val tpe = if (isArray) s"List[$enumName]" else enumName
+      val desc = param.description.map(d => JavaEscape.escapeString(d)).fold("")(d => s""".description("$d")""")
+      s""".in(${param.in}[$tpe]("${param.name}")$desc)""" -> Some(enumDefn)
+    }
     // .in(query[Limit]("limit").description("Maximum number of books to retrieve"))
     // .in(header[AuthToken]("X-Auth-Token"))
-    val params = parameters
+    val (params, maybeEnumDefns) = parameters
       .filter(_.in != "path")
       .map { param =>
         param.schema match {
           case st: OpenapiSchemaSimpleType =>
             val (t, _) = mapSchemaSimpleTypeToType(st)
             val req = if (param.required.getOrElse(true)) t else s"Option[$t]"
             val desc = param.description.map(d => JavaEscape.escapeString(d)).fold("")(d => s""".description("$d")""")
-            s""".in(${param.in}[$req]("${param.name}")$desc)"""
-          case x => bail(s"Can't create non-simple params to input - found $x")
+            s""".in(${param.in}[$req]("${param.name}")$desc)""" -> None
+          case OpenapiSchemaArray(st: OpenapiSchemaSimpleType, _) =>
+            val (t, _) = mapSchemaSimpleTypeToType(st)
+            val desc = param.description.map(d => JavaEscape.escapeString(d)).fold("")(d => s""".description("$d")""")
```

Comment:
Support for optional comma separated query param lists? 🙏 
```suggestion
            val arr = s"List[$t]"
            val req = if (param.required.getOrElse(true)) arr else s"Option[$arr]"
            val desc = param.description.map(d => JavaEscape.escapeString(d)).fold("")(d => s""".description("$d")""")
```

Reply:
This is much more involved than it seemed on the surface. Have reworked this pr a bit to also support exploded/unexploded query params to retain consistency between seq treatment of enum/not-enum